ci/scenarios/edge/main.py

```diff
@@ -0,0 +1,557 @@
+                    raise
+                finally:
+                    # Wait again to sync so we can properly abort the threads
+                    EdgeTester.LOGGER.info('Syncing threads')
+                    while r_semaphore.get_counter() < required_thread_amount:
+                        time.sleep(0.05)
+                EdgeTester._validate_dal(values_to_check)
+            except Exception as ex:
+                # @ TODO remove raise
+                raise
+                logger.error('Got an exception while running configuration {}. Namely: {}'.format(configuration, str(ex)))
+                failed_configurations.append({'configuration': configuration, 'reason': str(ex)})
+            finally:
+                if adjusted is True:
+                    EdgeTester.adjust_for_rerout(std_2.storagerouter, trigger_rerout=False, ip_to_block=compute_client.ip)
+                if screen_names:
+                    for screen_name in screen_names:
+                        pass  #@ Todo uncomment and remove pass
+                        # compute_client.run(['screen', '-S', screen_name, '-X', 'quit'])
+                try:
+                    logger.info('Stopping iops monitoring')
+                    for thread_pair in threads:
+                        if thread_pair[0].isAlive():
+                            thread_pair[1].set()
+                    if r_semaphore:
+                        r_semaphore.wait()
+                    # Wait for threads to die
+                    for thread_pair in threads:
+                        thread_pair[0].join()
+                except Exception as ex:
+                    logger.warning('Stopping the threads failed. Got {0}'.format(str(ex)))
+                if failed_over:
+                    # Wait for the downed node to come back up
+                    to_be_downed_client = None
+                    while to_be_downed_client is None:
+                        try:
+                            to_be_downed_client = SSHClient(str_2, username='root')
+                        except:
+                            pass
+                        time.sleep(1)
+                    services = ServiceManager.list_services(to_be_downed_client)
+                    for service in services:
+                        if vpool.name not in service:
+                            continue
+                        while ServiceManager.get_service_status(service, to_be_downed_client)[0] is False:
+                            time.sleep(1)
+                    for vdisk_name, vdisk_object in vdisk_info.iteritems():
+                        VDiskSetup.move_vdisk(vdisk_guid=vdisk_object.guid, target_storagerouter_guid=str_2.guid, api=api)
+
+        assert len(failed_configurations) == 0, 'Certain configuration failed: {0}'.format(failed_configurations)
+
+    @staticmethod
+    def _write_data(client, cmd_type, configuration, edge_configuration=None, screen=True, data_to_write=AMOUNT_TO_WRITE,
```

Comment:
we could make this part of a helper library so we can reuse this code in other tests?